// @engo/engo-ui-kit-lib/esm2022/lib/uik-table-editmode/uik-table-editmode.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { UikTableEditmodeComponent } from './uik-table-editmode.component';
import { MatPaginatorModule } from '@angular/material/paginator';
import { MatTableModule } from '@angular/material/table';
import { MatSortModule } from "@angular/material/sort";
import { UikStatusLabelModule } from "../uik-status-label/uik-status-label.module";
import * as i0 from "@angular/core";
export class UikTableEditmodeModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikTableEditmodeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.12", ngImport: i0, type: UikTableEditmodeModule, declarations: [UikTableEditmodeComponent], imports: [CommonModule,
            MatTableModule,
            MatSortModule,
            MatPaginatorModule,
            UikStatusLabelModule], exports: [UikTableEditmodeComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikTableEditmodeModule, imports: [CommonModule,
            MatTableModule,
            MatSortModule,
            MatPaginatorModule,
            UikStatusLabelModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikTableEditmodeModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        UikTableEditmodeComponent
                    ],
                    imports: [
                        CommonModule,
                        MatTableModule,
                        MatSortModule,
                        MatPaginatorModule,
                        UikStatusLabelModule
                    ],
                    exports: [
                        UikTableEditmodeComponent
                    ]
                }]
        }] });
