// @engo/engo-ui-kit-lib/esm2022/lib/uik-table-filter-menu/uik-table-filter-menu.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { UikTableFilterMenuComponent } from "./uik-table-filter-menu.component";
import { MatSlideToggleModule } from "@angular/material/slide-toggle";
import { MatMenuModule } from "@angular/material/menu";
import { MatBadgeModule } from "@angular/material/badge";
import * as i0 from "@angular/core";
export class UikTableFilterMenuModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikTableFilterMenuModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.12", ngImport: i0, type: UikTableFilterMenuModule, declarations: [UikTableFilterMenuComponent], imports: [CommonModule,
            MatSlideToggleModule,
            MatBadgeModule,
            MatMenuModule], exports: [UikTableFilterMenuComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikTableFilterMenuModule, imports: [CommonModule,
            MatSlideToggleModule,
            MatBadgeModule,
            MatMenuModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikTableFilterMenuModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        UikTableFilterMenuComponent
                    ],
                    exports: [
                        UikTableFilterMenuComponent
                    ],
                    imports: [
                        CommonModule,
                        MatSlideToggleModule,
                        MatBadgeModule,
                        MatMenuModule
                    ]
                }]
        }] });
