// @engo/engo-ui-kit-lib/esm2022/lib/uik-table-loading/uik-table-loading.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { UikTableLoadingComponent } from './uik-table-loading.component';
import { SkeletonModule } from "primeng/skeleton";
import * as i0 from "@angular/core";
export class UikTableLoadingModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikTableLoadingModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.12", ngImport: i0, type: UikTableLoadingModule, declarations: [UikTableLoadingComponent], imports: [CommonModule,
            SkeletonModule], exports: [UikTableLoadingComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikTableLoadingModule, imports: [CommonModule,
            SkeletonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikTableLoadingModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        UikTableLoadingComponent
                    ],
                    imports: [
                        CommonModule,
                        SkeletonModule
                    ],
                    exports: [
                        UikTableLoadingComponent
                    ]
                }]
        }] });
