// @engo/engo-ui-kit-lib/esm2022/lib/uik-table-checkbox/uik-table-checkbox.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { UikTableCheckboxComponent } from './uik-table-checkbox.component';
import { MatTableModule } from '@angular/material/table';
import { MatPaginatorModule } from '@angular/material/paginator';
import { MatCheckboxModule } from '@angular/material/checkbox';
import { MatSortModule } from "@angular/material/sort";
import { UikStatusLabelModule } from "../uik-status-label/uik-status-label.module";
import * as i0 from "@angular/core";
export class UikTableCheckboxModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikTableCheckboxModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.12", ngImport: i0, type: UikTableCheckboxModule, declarations: [UikTableCheckboxComponent], imports: [CommonModule,
            MatTableModule,
            MatSortModule,
            MatPaginatorModule,
            MatCheckboxModule,
            UikStatusLabelModule], exports: [UikTableCheckboxComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikTableCheckboxModule, imports: [CommonModule,
            MatTableModule,
            MatSortModule,
            MatPaginatorModule,
            MatCheckboxModule,
            UikStatusLabelModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikTableCheckboxModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        UikTableCheckboxComponent
                    ],
                    imports: [
                        CommonModule,
                        MatTableModule,
                        MatSortModule,
                        MatPaginatorModule,
                        MatCheckboxModule,
                        UikStatusLabelModule
                    ],
                    exports: [
                        UikTableCheckboxComponent
                    ]
                }]
        }] });
