// @engo/engo-ui-kit-lib/esm2022/lib/uik-forms/uik-form-legend/uik-form-legend.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { UikFormLegendComponent } from './uik-form-legend.component';
import * as i0 from "@angular/core";
export class UikFormLegendModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikFormLegendModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.12", ngImport: i0, type: UikFormLegendModule, declarations: [UikFormLegendComponent], imports: [CommonModule], exports: [UikFormLegendComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikFormLegendModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikFormLegendModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        UikFormLegendComponent
                    ],
                    imports: [
                        CommonModule
                    ],
                    exports: [
                        UikFormLegendComponent
                    ]
                }]
        }] });
