// @engo/engo-ui-kit-lib/esm2022/lib/uik-status-label/uik-status-label.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { UikStatusLabelComponent } from './uik-status-label.component';
import * as i0 from "@angular/core";
export class UikStatusLabelModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikStatusLabelModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.12", ngImport: i0, type: UikStatusLabelModule, declarations: [UikStatusLabelComponent], imports: [CommonModule], exports: [UikStatusLabelComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikStatusLabelModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikStatusLabelModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        UikStatusLabelComponent
                    ],
                    exports: [
                        UikStatusLabelComponent
                    ],
                    imports: [
                        CommonModule
                    ]
                }]
        }] });
