// @engo/engo-ui-kit-lib/esm2022/lib/uik-confirmation-dialog/uik-confirmation-dialog.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { UikConfirmationDialogComponent } from './uik-confirmation-dialog.component';
import { MatDialogModule } from '@angular/material/dialog';
import * as i0 from "@angular/core";
export class UikConfirmationDialogModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikConfirmationDialogModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.12", ngImport: i0, type: UikConfirmationDialogModule, declarations: [UikConfirmationDialogComponent], imports: [CommonModule,
            MatDialogModule], exports: [UikConfirmationDialogComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikConfirmationDialogModule, imports: [CommonModule,
            MatDialogModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikConfirmationDialogModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        UikConfirmationDialogComponent
                    ],
                    imports: [
                        CommonModule,
                        MatDialogModule,
                    ],
                    exports: [
                        UikConfirmationDialogComponent
                    ]
                }]
        }] });
