// @engo/engo-ui-kit-lib/esm2022/lib/uik-confirmation-dialog/uik-confirmation-dialog.component.mjs
import { Component, Inject } from '@angular/core';
import { MAT_DIALOG_DATA } from '@angular/material/dialog';
import * as i0 from "@angular/core";
import * as i1 from "@angular/material/dialog";
export class UikConfirmationDialogComponent {
    constructor(data, dialogRef) {
        this.dialogRef = dialogRef;
        this.message = 'Are you sure?';
        this.confirmButtonText = 'Yes';
        this.cancelButtonText = 'Cancel';
        this.confirmFaIconClass = 'fa fa-trash';
        this.cancelFaIconClass = 'fa fa-ban';
        if (data) {
            this.message = data.message || this.message;
            if (data.buttonText) {
                this.confirmButtonText = data.buttonText.ok || this.confirmButtonText;
                this.cancelButtonText = data.buttonText.cancel || this.cancelButtonText;
            }
            if (data.buttonIcons) {
                this.confirmFaIconClass = data.buttonIcons.ok || this.confirmFaIconClass;
                this.cancelFaIconClass = data.buttonIcons.cancel || this.cancelFaIconClass;
            }
        }
    }
    ngOnInit() {
    }
    onConfirmClick() {
        this.dialogRef.close(true);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikConfirmationDialogComponent, deps: [{ token: MAT_DIALOG_DATA }, { token: i1.MatDialogRef }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: UikConfirmationDialogComponent, selector: "engo-uik-confirmation-dialog", ngImport: i0, template: "<mat-dialog-content class=\"confirmation-dialog-content\">\n  <p>\n    {{message}}\n  </p>\n</mat-dialog-content>\n<mat-dialog-actions>\n  <button class=\"btn-action-button btn-gray\" mat-dialog-close><i\n    [class]=\"cancelFaIconClass + ' btn-icon'\"></i>{{cancelButtonText}}</button>\n\n  <button (click)=\"onConfirmClick()\" class=\"btn-action-button\"><i\n    [class]=\"confirmFaIconClass + ' btn-icon'\"></i>{{confirmButtonText}}</button>\n</mat-dialog-actions>\n", styles: [".mat-mdc-dialog-actions{display:flex;justify-content:right;margin:12px;flex-wrap:wrap}.btn-action-button{margin-right:12px}.confirmation-dialog-content{max-width:1000px}\n"], dependencies: [{ kind: "directive", type: i1.MatDialogClose, selector: "[mat-dialog-close], [matDialogClose]", inputs: ["aria-label", "type", "mat-dialog-close", "matDialogClose"], exportAs: ["matDialogClose"] }, { kind: "directive", type: i1.MatDialogContent, selector: "[mat-dialog-content], mat-dialog-content, [matDialogContent]" }, { kind: "directive", type: i1.MatDialogActions, selector: "[mat-dialog-actions], mat-dialog-actions, [matDialogActions]", inputs: ["align"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikConfirmationDialogComponent, decorators: [{
            type: Component,
            args: [{ selector: 'engo-uik-confirmation-dialog', template: "<mat-dialog-content class=\"confirmation-dialog-content\">\n  <p>\n    {{message}}\n  </p>\n</mat-dialog-content>\n<mat-dialog-actions>\n  <button class=\"btn-action-button btn-gray\" mat-dialog-close><i\n    [class]=\"cancelFaIconClass + ' btn-icon'\"></i>{{cancelButtonText}}</button>\n\n  <button (click)=\"onConfirmClick()\" class=\"btn-action-button\"><i\n    [class]=\"confirmFaIconClass + ' btn-icon'\"></i>{{confirmButtonText}}</button>\n</mat-dialog-actions>\n", styles: [".mat-mdc-dialog-actions{display:flex;justify-content:right;margin:12px;flex-wrap:wrap}.btn-action-button{margin-right:12px}.confirmation-dialog-content{max-width:1000px}\n"] }]
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [MAT_DIALOG_DATA]
                }] }, { type: i1.MatDialogRef }]; } });
