// @engo/engo-ui-kit-lib/esm2022/lib/uik-form-loading/uik-form-loading.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { UikFormLoadingComponent } from './uik-form-loading.component';
import { SkeletonModule } from "primeng/skeleton";
import * as i0 from "@angular/core";
export class UikFormLoadingModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikFormLoadingModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.12", ngImport: i0, type: UikFormLoadingModule, declarations: [UikFormLoadingComponent], imports: [CommonModule,
            SkeletonModule], exports: [UikFormLoadingComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikFormLoadingModule, imports: [CommonModule,
            SkeletonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikFormLoadingModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        UikFormLoadingComponent
                    ],
                    imports: [
                        CommonModule,
                        SkeletonModule
                    ],
                    exports: [
                        UikFormLoadingComponent
                    ]
                }]
        }] });
