// @hreimer/angular-image-viewer/esm2020/lib/angular-image-viewer.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { AngularImageViewerComponent } from './angular-image-viewer.component';
import { FullScreenDirective } from './directives/full-screen.directive';
import * as i0 from "@angular/core";
// import { ImageViewerConfig } from 'dist/angular-image-viewer/public-api';
export class AngularImageViewerModule {
}
AngularImageViewerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.3.0", ngImport: i0, type: AngularImageViewerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AngularImageViewerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.3.0", ngImport: i0, type: AngularImageViewerModule, declarations: [AngularImageViewerComponent, FullScreenDirective], imports: [CommonModule,
        BrowserModule,
        FormsModule,
        BrowserAnimationsModule], exports: [AngularImageViewerComponent, FullScreenDirective] });
AngularImageViewerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.3.0", ngImport: i0, type: AngularImageViewerModule, imports: [CommonModule,
        BrowserModule,
        FormsModule,
        BrowserAnimationsModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.3.0", ngImport: i0, type: AngularImageViewerModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [AngularImageViewerComponent, FullScreenDirective],
                    imports: [
                        CommonModule,
                        BrowserModule,
                        FormsModule,
                        BrowserAnimationsModule,
                    ],
                    exports: [AngularImageViewerComponent, FullScreenDirective]
                }]
        }] });
