// @engo/engo-ui-kit-lib/esm2022/lib/uik-routing-loading-bar/uik-routing-loading-bar.component.mjs
import { Component } from '@angular/core';
import { NavigationCancel, NavigationEnd, NavigationStart } from "@angular/router";
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
import * as i2 from "@angular/common";
import * as i3 from "@angular/material/progress-bar";
export class UikRoutingLoadingBarComponent {
    constructor(router) {
        this.router = router;
        this.loading = false;
    }
    ngOnInit() {
        this.router.events.subscribe((event) => {
            switch (true) {
                case event instanceof NavigationStart: {
                    this.loading = true;
                    break;
                }
                case event instanceof NavigationEnd: {
                    this.loading = false;
                    break;
                }
                case event instanceof NavigationCancel: {
                    this.loading = false;
                    break;
                }
            }
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikRoutingLoadingBarComponent, deps: [{ token: i1.Router }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: UikRoutingLoadingBarComponent, selector: "engo-uik-routing-loading-bar", ngImport: i0, template: "<mat-progress-bar *ngIf=\"loading\" [mode]=\"'indeterminate'\"></mat-progress-bar>\n", styles: [".mat-mdc-progress-bar{width:102%!important;margin-left:-1%!important}\n"], dependencies: [{ kind: "directive", type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "component", type: i3.MatProgressBar, selector: "mat-progress-bar", inputs: ["color", "value", "bufferValue", "mode"], outputs: ["animationEnd"], exportAs: ["matProgressBar"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikRoutingLoadingBarComponent, decorators: [{
            type: Component,
            args: [{ selector: 'engo-uik-routing-loading-bar', template: "<mat-progress-bar *ngIf=\"loading\" [mode]=\"'indeterminate'\"></mat-progress-bar>\n", styles: [".mat-mdc-progress-bar{width:102%!important;margin-left:-1%!important}\n"] }]
        }], ctorParameters: function () { return [{ type: i1.Router }]; } });
