// @engo/engo-ui-kit-lib/esm2022/lib/uik-routing-loading-bar/uik-routing-loading-bar.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { UikRoutingLoadingBarComponent } from './uik-routing-loading-bar.component';
import { MatProgressBarModule } from "@angular/material/progress-bar";
import * as i0 from "@angular/core";
export class UikRoutingLoadingBarModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikRoutingLoadingBarModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.12", ngImport: i0, type: UikRoutingLoadingBarModule, declarations: [UikRoutingLoadingBarComponent], imports: [CommonModule,
            MatProgressBarModule], exports: [UikRoutingLoadingBarComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikRoutingLoadingBarModule, imports: [CommonModule,
            MatProgressBarModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikRoutingLoadingBarModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        UikRoutingLoadingBarComponent
                    ],
                    exports: [
                        UikRoutingLoadingBarComponent
                    ],
                    imports: [
                        CommonModule,
                        MatProgressBarModule
                    ]
                }]
        }] });
