// @engo/engo-ui-kit-lib/esm2022/lib/uik-table-more-actions/uik-table-more-actions.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { UikTableMoreActions } from './uik-table-more-actions.component';
import { MatMenuModule } from "@angular/material/menu";
import * as i0 from "@angular/core";
export class UikTableMoreActionsModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikTableMoreActionsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.12", ngImport: i0, type: UikTableMoreActionsModule, declarations: [UikTableMoreActions], imports: [CommonModule,
            MatMenuModule], exports: [UikTableMoreActions] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikTableMoreActionsModule, imports: [CommonModule,
            MatMenuModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikTableMoreActionsModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        UikTableMoreActions
                    ],
                    exports: [
                        UikTableMoreActions
                    ],
                    imports: [
                        CommonModule,
                        MatMenuModule
                    ]
                }]
        }] });
