// @engo/engo-ui-kit-lib/esm2022/lib/uik-forms/uik-forms/uik-forms.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { UikFormsComponent } from './uik-forms.component';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { MatFormFieldModule } from '@angular/material/form-field';
import { MatSelectModule } from '@angular/material/select';
import { NgSelectModule } from '@ng-select/ng-select';
import * as i0 from "@angular/core";
export class UikFormsModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikFormsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.12", ngImport: i0, type: UikFormsModule, declarations: [UikFormsComponent], imports: [CommonModule,
            FormsModule,
            MatFormFieldModule,
            MatSelectModule,
            ReactiveFormsModule,
            NgSelectModule], exports: [UikFormsComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikFormsModule, imports: [CommonModule,
            FormsModule,
            MatFormFieldModule,
            MatSelectModule,
            ReactiveFormsModule,
            NgSelectModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikFormsModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        UikFormsComponent
                    ],
                    imports: [
                        CommonModule,
                        FormsModule,
                        MatFormFieldModule,
                        MatSelectModule,
                        ReactiveFormsModule,
                        NgSelectModule
                    ],
                    exports: [
                        UikFormsComponent
                    ]
                }]
        }] });
