// @engo/engo-ui-kit-lib/esm2022/lib/uik-table/uik-table.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { UikTableComponent } from './uik-table.component';
import { MatTableModule } from '@angular/material/table';
import { MatPaginatorModule } from '@angular/material/paginator';
import { MatSortModule } from '@angular/material/sort';
import { UikStatusLabelModule } from '../uik-status-label/uik-status-label.module';
import * as i0 from "@angular/core";
export class UikTableModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikTableModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.12", ngImport: i0, type: UikTableModule, declarations: [UikTableComponent], imports: [CommonModule,
            MatTableModule,
            MatSortModule,
            MatPaginatorModule,
            UikStatusLabelModule], exports: [UikTableComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikTableModule, imports: [CommonModule,
            MatTableModule,
            MatSortModule,
            MatPaginatorModule,
            UikStatusLabelModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikTableModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [UikTableComponent],
                    imports: [
                        CommonModule,
                        MatTableModule,
                        MatSortModule,
                        MatPaginatorModule,
                        UikStatusLabelModule
                    ],
                    exports: [
                        UikTableComponent
                    ]
                }]
        }] });
