// @engo/engo-ui-kit-lib/esm2022/lib/uik-table-filter-by-list-menu/uik-table-filter-by-list-menu.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { UikTableFilterByListMenuComponent } from './uik-table-filter-by-list-menu.component';
import { MatMenuModule } from "@angular/material/menu";
import { MatSelectModule } from "@angular/material/select";
import { MatBadgeModule } from "@angular/material/badge";
import { DropdownModule } from "primeng/dropdown";
import { FormsModule } from "@angular/forms";
import { MultiSelectModule } from "primeng/multiselect";
import * as i0 from "@angular/core";
export class UikTableFilterByListMenuModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikTableFilterByListMenuModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.12", ngImport: i0, type: UikTableFilterByListMenuModule, declarations: [UikTableFilterByListMenuComponent], imports: [CommonModule,
            MatMenuModule,
            MatSelectModule,
            MatBadgeModule,
            DropdownModule,
            FormsModule,
            MultiSelectModule], exports: [UikTableFilterByListMenuComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikTableFilterByListMenuModule, imports: [CommonModule,
            MatMenuModule,
            MatSelectModule,
            MatBadgeModule,
            DropdownModule,
            FormsModule,
            MultiSelectModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: UikTableFilterByListMenuModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        UikTableFilterByListMenuComponent
                    ],
                    exports: [
                        UikTableFilterByListMenuComponent
                    ],
                    imports: [
                        CommonModule,
                        MatMenuModule,
                        MatSelectModule,
                        MatBadgeModule,
                        DropdownModule,
                        FormsModule,
                        MultiSelectModule
                    ]
                }]
        }] });
